lib/components/user/favorite-locations-pane.js

```diff
@@ -1,151 +1,48 @@
-import { Field, FieldArray } from 'formik'
-import memoize from 'lodash.memoize'
-import React, { Component } from 'react'
-import {
-  Button,
-  ControlLabel,
-  FormControl,
-  FormGroup,
-  InputGroup
-} from 'react-bootstrap'
-import FontAwesome from 'react-fontawesome'
-import styled from 'styled-components'
+import { FieldArray } from 'formik'
+import React from 'react'
+import { ControlLabel } from 'react-bootstrap'
 
-// Styles.
-const fancyAddLocationCss = `
-  background-color: #337ab7;
-  color: #fff;
-`
-const StyledAddon = styled(InputGroup.Addon)`
-  min-width: 40px;
-`
-const NewLocationAddon = styled(StyledAddon)`
-  ${fancyAddLocationCss}
-`
-const NewLocationFormControl = styled(FormControl)`
-  ${fancyAddLocationCss}
-  ::placeholder {
-    color: #fff;
-  }
-  &:focus {
-    background-color: unset;
-    color: unset;
-    ::placeholder {
-      color: unset;
-    }
-  }
-`
-
-// Helper filter functions.
-export const isHome = loc => loc.type === 'home'
-export const isWork = loc => loc.type === 'work'
+import FavoriteLocation, { isHome, isWork } from './favorite-location'
 
 /**
- * Helper function that adds a new address to the Formik state
- * using the Formik-provided arrayHelpers object.
+ * Renders an editable list user's favorite locations, and lets the user add a new one.
  */
-function addNewAddress (arrayHelpers, e) {
-  const value = (e.target.value || '').trim()
-  if (value.length > 0) {
-    arrayHelpers.push({
-      address: value,
-      icon: 'map-marker',
-      type: 'custom'
-    })
-
-    // Empty the input box value so the user can enter their next location.
-    e.target.value = ''
-  }
-}
-
-/**
- * User's saved locations editor.
- * TODO: Discuss and improve handling of location details (type, coordinates...).
- */
-class FavoriteLocationsPane extends Component {
-  _handleNewAddressKeyDown = memoize(
-    arrayHelpers => e => {
-      if (e.keyCode === 13) {
-        // On the user pressing enter (keyCode 13) on the new location input,
-        // add new address to user's savedLocations...
-        addNewAddress(arrayHelpers, e)
-
-        // ... but don't submit the form.
-        e.preventDefault()
-      }
-    }
+const FavoriteLocationsPane = ({ values: userData }) => {
+  const { savedLocations } = userData
+  const homeLocation = savedLocations.find(isHome)
+  const workLocation = savedLocations.find(isWork)
+
+  return (
+    <div>
+      <ControlLabel>Add the places you frequent often to save time planning trips:</ControlLabel>
+
+      <FieldArray
+        name='savedLocations'
+        render={arrayHelpers => (
+          <>
+            {savedLocations.map((loc, index) => {
+              const isHomeOrWork = loc === homeLocation || loc === workLocation
```

Comment:
No need to define `homeLocation` and `workLocation` variables, right? I would rather see the `isFixed` prop go away entirely and just have favorite-location handle this internally (no need to define this extra prop as far as I can tell).

Reply:
Hmmm no, this would couple FavoriteLocations with isHome and isWork.

Reply:
FavoriteLocations is tied with isHome and isWork right now, but I might have an idea.

Reply:
Okay, I have simplified the overall handling of work and home in https://github.com/opentripplanner/otp-react-redux/pull/280/commits/bcce0e4b6ce0cc0606208d6eac75dc7380fd8f77.

Reply:
I eventually did away with that `isFixed` or `placeholder` prop. There is a new component `FixedFavoriteLocation` from https://github.com/opentripplanner/otp-react-redux/pull/280/commits/b5b3739a2a63157c758ea734c3e8507793f94cfc.